powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/v3/PrepareActivationStep.java

```diff
@@ -198,17 +198,18 @@ public JSONObject execute(JsonStepLogger stepLogger, Map<String, Object> context
                     .asString();
 
             if (response.getStatus() == 200) {
-                // Read activation layer 1 response and decrypt it
                 EciesEncryptedResponse encryptedResponseL1 = objectMapper.readValue(response.getRawBody(), EciesEncryptedResponse.class);
+
+                if (stepLogger != null) {
+                    stepLogger.writeServerCallOK(encryptedResponseL1, HttpUtil.flattenHttpHeaders(response.getHeaders()));
+                }
+
+                // Read activation layer 1 response and decrypt it
                 byte[] macL1 = BaseEncoding.base64().decode(encryptedResponseL1.getMac());
                 byte[] encryptedDataL1 = BaseEncoding.base64().decode(encryptedResponseL1.getEncryptedData());
                 EciesCryptogram responseCryptogramL1 = new EciesCryptogram(macL1, encryptedDataL1);
                 byte[] decryptedDataL1 = eciesEncryptorL1.decryptResponse(responseCryptogramL1);
 
-                if (stepLogger != null) {
```

Comment:
Why are we removing this logging point? We would like to log decrypted data as well, right? Maybe using `writeServerCallOK` is not the best way of logging, but we should log the original request / response...